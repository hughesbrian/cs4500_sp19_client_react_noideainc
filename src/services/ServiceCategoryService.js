const baseURL = "https://cs4500-sp19-noideainc.herokuapp.com";
//const baseURL = "http://localhost:8080";

export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if (ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }

    findServiceCategoryById = categoryId =>
        fetch(baseURL + `/api/categories/${categoryId}`)
            .then(response => response.json())

    findAllServiceCategories = () =>
        fetch(baseURL +  `/api/categories`)
            .then(response => response.json())

    pagedServiceCategories = (page, count) =>
        fetch(baseURL + `/api/categories/paged?page=${page}&count=${count}`)
            .then(response => response.json())

    createServiceCategory = sc => {
        delete sc.id;
        return fetch(baseURL + `/api/categories`, {
            method: 'post',
            body: JSON.stringify(sc),
            headers: {
                'content-type': 'application/json'
            }
        })
        .then(response => response.json())
    }

    updateServiceCategory = sc =>
        fetch(baseURL + `/api/categories/${sc.id}`, {
            method: 'put',
            body: JSON.stringify(sc),
            headers: {
                'content-type': 'application/json'
            }
        }).then(response => response.json())
    
    updateServiceCategoryScore = sc =>
        fetch(baseURL + `/api/categories/score/${sc.id}`, {
            method: 'put',
            body: JSON.stringify(sc),
            headers: {
                'content-type': 'application/json'
            }
        }).then(response => response.json())

    deleteServiceCategory = id =>
        fetch(baseURL + `/api/categories/${id}`, {
            method: 'delete'
        })

    findAllServicesByCategoryName = catName =>
        fetch(baseURL +  `/api/${catName}`)
            .then(response => response.json())
}
