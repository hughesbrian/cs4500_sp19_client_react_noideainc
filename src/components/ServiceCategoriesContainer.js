import React from 'react'
import ServiceCategories from './ServiceCategories'
import ServiceCategoryService from '../services/ServiceCategoryService'

class ServiceCategoriesContainer extends React.Component {
    constructor(props) {
        super(props);
        this.serviceCategoryService = ServiceCategoryService.getInstance();
        this.state = {
            page: 0,
            count: 10,
            totalPages: 1,
            serviceCategories: [],
            serviceCategory: {
                title: '',
            }
        }
    }
    componentDidMount() {
        this.findAllServiceCategories()
    }
    findAllServiceCategories = () =>
        this.serviceCategoryService
            .pagedServiceCategories(this.state.page, this.state.count)
            .then(response => {
                this.setState({
                    serviceCategories: response.content,
                    totalPages: response.totalPages
                })
            })
    updateForm = e =>
        this.setState({
            serviceCategory: {
                id: this.state.serviceCategory.id,
                title: e.target.value
            }
        })
    createServiceCategory = () => {
        this.serviceCategoryService
            .createServiceCategory(this.state.serviceCategory)
            .then(this.findAllServiceCategories)
    }
    deleteServiceCategory = id =>
        this.serviceCategoryService
            .deleteServiceCategory(id)
            .then(this.findAllServiceCategories)
    selectServiceCategory = id =>
        this.serviceCategoryService
            .findServiceCategoryById(id)
            .then(sc => this.setState({
                serviceCategory: sc
            }))
    updateServiceCategory = () =>
        this.serviceCategoryService
            .updateServiceCategory(this.state.serviceCategory)
            .then(this.findAllServiceCategories)
    setPageSelect = () => {
        this.state.page = document.getElementById("select-page").value
        this.findAllServiceCategories()
    }
    setPagePagination = page => {
        if(page < this.state.totalPages && page >= 0) {
            this.state.page = page
            this.findAllServiceCategories()
        }
    }

    render = () =>
        <ServiceCategories page={this.state.page}
                           count={this.state.count}
                           totalPages={this.state.totalPages}
                           serviceCategories={this.state.serviceCategories}
                           serviceCategory={this.state.serviceCategory}
                           updateForm={this.updateForm}
                           createServiceCategory={this.createServiceCategory}
                           deleteServiceCategory={this.deleteServiceCategory}
                           selectServiceCategory={this.selectServiceCategory}
                           updateServiceCategory={this.updateServiceCategory}
                           setPageSelect={this.setPageSelect}
                           setPagePagination={this.setPagePagination}/>
}

export default ServiceCategoriesContainer
