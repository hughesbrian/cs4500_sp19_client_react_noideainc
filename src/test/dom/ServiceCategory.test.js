import React from 'react'
import ServiceCategoriesContainer from '../../components/ServiceCategoriesContainer'
import TestRenderer from 'react-test-renderer'
import serviceCategories from '../mockdata/ServiceCategories.mock.json'
import sleep from '../util/sleep'
import '../mockservice/ServiceCategoryService.mock'

test('ServiceCategoriesContainer renders correctly', async () => {
    const testRenderer = TestRenderer.create(
        <ServiceCategoriesContainer/>
    )
    await sleep(1000) // sleep to give the component a chance to render.

    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance = testRenderer.root

    const selectBtns = testInstance.findAllByProps({className: 'btn btn-primary select-btn'})
    await sleep(500) // sleep to give the component a chance to render.
    selectBtns[0].props.onClick()
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const titleField = testInstance.findByProps({className: 'title-field'})
    await sleep(500) // sleep to give the component a chance to render.
    titleField.props.onChange({target: {value: "New category"}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const updateBtn = testInstance.findByProps({className: 'btn btn-success update-btn'})
    await sleep(500) // sleep to give the component a chance to render.
    updateBtn.props.onClick()
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    await sleep(500) // sleep to give the component a chance to render.
    titleField.props.onChange({target: {value: "Creation"}})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const createBtn = testInstance.findByProps({className: 'btn btn-primary create-btn'})
    await sleep(500) // sleep to give the component a chance to render.
    createBtn.props.onClick()
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const deleteBtns = testInstance.findAllByProps({className: 'btn btn-danger delete-btn'})
    await sleep(500) // sleep to give the component a chance to render.
    const last = deleteBtns.length - 1
    deleteBtns[last].props.onClick(serviceCategories[last].id)
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})
