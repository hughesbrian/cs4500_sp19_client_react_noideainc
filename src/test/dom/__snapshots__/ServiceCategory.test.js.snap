// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCategoriesContainer renders correctly 1`] = `
<div>
  <h3>
    Service Categories
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <b>
            Title
          </b>
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="title-field"
            onChange={[Function]}
            value=""
          />
        </td>
        <td>
          <button
            className="btn btn-primary create-btn"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="btn btn-success update-btn"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Tutoring
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Fitness Trainer
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          House Painter
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-3"
            >
              <select
                className="form-control"
                id="select-page"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="page-select-fields"
                  value={0}
                >
                  1
                </option>
              </select>
            </div>
            <div
              className="col-9"
            >
              <nav
                aria-label="Page navigation example"
              >
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link previous-btn"
                      onClick={[Function]}
                    >
                      Previous
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link pagination-btns"
                      onClick={[Function]}
                    >
                      1
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link next-btn"
                      onClick={[Function]}
                    >
                      Next
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ServiceCategoriesContainer renders correctly 2`] = `
<div>
  <h3>
    Service Categories
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <b>
            Title
          </b>
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="title-field"
            onChange={[Function]}
            value=""
          />
        </td>
        <td>
          <button
            className="btn btn-primary create-btn"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="btn btn-success update-btn"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Tutoring
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Fitness Trainer
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          House Painter
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-3"
            >
              <select
                className="form-control"
                id="select-page"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="page-select-fields"
                  value={0}
                >
                  1
                </option>
              </select>
            </div>
            <div
              className="col-9"
            >
              <nav
                aria-label="Page navigation example"
              >
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link previous-btn"
                      onClick={[Function]}
                    >
                      Previous
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link pagination-btns"
                      onClick={[Function]}
                    >
                      1
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link next-btn"
                      onClick={[Function]}
                    >
                      Next
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ServiceCategoriesContainer renders correctly 3`] = `
<div>
  <h3>
    Service Categories
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <b>
            Title
          </b>
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="title-field"
            onChange={[Function]}
            value="New category"
          />
        </td>
        <td>
          <button
            className="btn btn-primary create-btn"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="btn btn-success update-btn"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Tutoring
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Fitness Trainer
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          House Painter
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-3"
            >
              <select
                className="form-control"
                id="select-page"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="page-select-fields"
                  value={0}
                >
                  1
                </option>
              </select>
            </div>
            <div
              className="col-9"
            >
              <nav
                aria-label="Page navigation example"
              >
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link previous-btn"
                      onClick={[Function]}
                    >
                      Previous
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link pagination-btns"
                      onClick={[Function]}
                    >
                      1
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link next-btn"
                      onClick={[Function]}
                    >
                      Next
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ServiceCategoriesContainer renders correctly 4`] = `
<div>
  <h3>
    Service Categories
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <b>
            Title
          </b>
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="title-field"
            onChange={[Function]}
            value="New category"
          />
        </td>
        <td>
          <button
            className="btn btn-primary create-btn"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="btn btn-success update-btn"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Tutoring
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Fitness Trainer
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          House Painter
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-3"
            >
              <select
                className="form-control"
                id="select-page"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="page-select-fields"
                  value={0}
                >
                  1
                </option>
              </select>
            </div>
            <div
              className="col-9"
            >
              <nav
                aria-label="Page navigation example"
              >
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link previous-btn"
                      onClick={[Function]}
                    >
                      Previous
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link pagination-btns"
                      onClick={[Function]}
                    >
                      1
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link next-btn"
                      onClick={[Function]}
                    >
                      Next
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ServiceCategoriesContainer renders correctly 5`] = `
<div>
  <h3>
    Service Categories
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <b>
            Title
          </b>
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="title-field"
            onChange={[Function]}
            value="Creation"
          />
        </td>
        <td>
          <button
            className="btn btn-primary create-btn"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="btn btn-success update-btn"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          New category
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Fitness Trainer
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          House Painter
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-3"
            >
              <select
                className="form-control"
                id="select-page"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="page-select-fields"
                  value={0}
                >
                  1
                </option>
              </select>
            </div>
            <div
              className="col-9"
            >
              <nav
                aria-label="Page navigation example"
              >
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link previous-btn"
                      onClick={[Function]}
                    >
                      Previous
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link pagination-btns"
                      onClick={[Function]}
                    >
                      1
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link next-btn"
                      onClick={[Function]}
                    >
                      Next
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ServiceCategoriesContainer renders correctly 6`] = `
<div>
  <h3>
    Service Categories
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <b>
            Title
          </b>
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="title-field"
            onChange={[Function]}
            value="Creation"
          />
        </td>
        <td>
          <button
            className="btn btn-primary create-btn"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="btn btn-success update-btn"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          New category
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Fitness Trainer
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          House Painter
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-3"
            >
              <select
                className="form-control"
                id="select-page"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="page-select-fields"
                  value={0}
                >
                  1
                </option>
              </select>
            </div>
            <div
              className="col-9"
            >
              <nav
                aria-label="Page navigation example"
              >
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link previous-btn"
                      onClick={[Function]}
                    >
                      Previous
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link pagination-btns"
                      onClick={[Function]}
                    >
                      1
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link next-btn"
                      onClick={[Function]}
                    >
                      Next
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ServiceCategoriesContainer renders correctly 7`] = `
<div>
  <h3>
    Service Categories
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <b>
            Title
          </b>
        </td>
      </tr>
      <tr>
        <td>
          <input
            className="title-field"
            onChange={[Function]}
            value="Creation"
          />
        </td>
        <td>
          <button
            className="btn btn-primary create-btn"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="btn btn-success update-btn"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          New category
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Fitness Trainer
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          House Painter
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-rows"
      >
        <td>
          Creation
        </td>
        <td>
          <button
            className="btn btn-danger delete-btn"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="btn btn-primary select-btn"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-3"
            >
              <select
                className="form-control"
                id="select-page"
                onChange={[Function]}
                value={0}
              >
                <option
                  className="page-select-fields"
                  value={0}
                >
                  1
                </option>
              </select>
            </div>
            <div
              className="col-9"
            >
              <nav
                aria-label="Page navigation example"
              >
                <ul
                  className="pagination"
                >
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link previous-btn"
                      onClick={[Function]}
                    >
                      Previous
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link pagination-btns"
                      onClick={[Function]}
                    >
                      1
                    </button>
                  </li>
                  <li
                    className="page-item"
                  >
                    <button
                      className="page-link next-btn"
                      onClick={[Function]}
                    >
                      Next
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
